src/lib/viewers/controls/page/PageControlsForm.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { decodeKeydown } from '../../../util';
+import './PageControlsForm.scss';
+
+export type Props = {
+    onPageSubmit: (newPageNumber: number) => void;
+    pageNumber: number;
+    pageCount: number;
+};
+
+export const ENTER = 'Enter';
+export const ESCAPE = 'Escape';
+
+export default function PageControlsForm({ onPageSubmit, pageNumber, pageCount }: Props): JSX.Element {
+    const [isInputShown, setIsInputShown] = React.useState(false);
+    const [inputValue, setInputValue] = React.useState(pageNumber.toString());
+    const [isButtonFocus, setIsButtonFocus] = React.useState(false);
+    const buttonElRef = React.useRef<HTMLButtonElement>(null);
+    const inputElRef = React.useRef<HTMLInputElement>(null);
+
+    const setPage = (allowRetry = false): void => {
+        const newPageNumber = parseInt(inputValue, 10);
+
+        if (
+            !Number.isNaN(newPageNumber) &&
```

Comment:
Would Number.isFinite work here instead of the negated isNaN?

Reply:
I believe that can work, but I'd rather use NaN to signify since it might be more readable.